lib/pacemaker/pcmk_sched_promotable.c

```diff
@@ -337,15 +407,21 @@ apply_promoted_dependent_colocations(pe_resource_t *rsc)
         }
 
         pe_rsc_trace(rsc, "RHS: %s with %s: %d", cons->rsc_lh->id,
-                     cons->rsc_rh->id, cons->score);
+                    cons->rsc_rh->id, cons->score);
 
-        rsc->allowed_nodes =
-            cons->rsc_rh->cmds->merge_weights(cons->rsc_rh, rsc->id,
-                                              rsc->allowed_nodes,
-                                              cons->node_attribute,
+        work_nodes =
+            cons->rsc_rh->cmds->merge_weights(cons->rsc_rh, work_rsc->id,
+                                              work_nodes, cons->node_attribute,
                                               (cons->score / (float) INFINITY),
                                               flags);
     }
+
+    if (bundled) {
+        apply_host_scores_to_bundle_nodes(rsc, work_nodes);
+        g_hash_table_destroy(work_nodes);
+    } else {
+        rsc->allowed_nodes = work_nodes;
```

Comment:
I'm making myself a note to re-write this `work_nodes` scheme in a better way (maybe using a `backup` approach instead). This is sloppy and probably leaves a memory leak. It was the last thing I did that night and it shows.

Reply:
Done